enhancements/storage/csi-driver-operator-merge.md

```diff
@@ -469,6 +469,80 @@ N/A
 
 Same as today.
 
+## Process of moving operators to csi-operator monorepo
+
+We have come with following flow for moving operators from their own repository into csi-operator mono repo.
+
+### Avoid .gitignore related footguns
+
+Remove any .gitignore entries in the source repository that would match a directory / file that we need. For example, `azure-disk-csi-driver-operator` in .gitignore matched `cmd/azure-disk-csi-driver-operator` directory that we really need not to be ignored. See https://github.com/openshift/csi-operator/pull/110, where we had to fix after merge to csi-operator.
+
+### Move existing code into csi-operator repository
+
+Using git-subtree move your existing operator code to https://github.com/openshift/csi-operator/tree/master/legacy
+
+```
+git subtree add --prefix legacy/azure-disk-csi-driver-operator https://github.com/openshift/azure-disk-csi-driver-operator.git master --squash
+git subtree push --prefix legacy/azure-disk-csi-driver-operator https://github.com/openshift/azure-disk-csi-driver-operator.git master
+```
+
+### Add Dockerfiles for building images from new location
+
+Place a `Dockerfile.<operator>` and `Dockerfile.<operator>.test` at top of csi-operator tree and make sure that you are able to build an image of the operator from csi-operator repository.
+
+### Update openshift/release to build image from new location
+Make a PR to openshift/release repository to build the operator from csi-operator. For example - https://github.com/openshift/release/pull/46233.
+
+1. Update also `storage-conf-csi-<operator>-commands.sh`, the test manifest will be at a different location.
+2. Make sure that rehearse jobs for both older versions of operator and newer versions of operator pass.
+
+### Change ocp-build-data repository to ship image from new location
+
+Make a PR to [ocp-build-data](https://github.com/openshift-eng/ocp-build-data) repository to change location of the image etc -  https://github.com/openshift-eng/ocp-build-data/pull/4148
+
+1. Notice the `cachito` line in the PR - we need to build with the vendor from legacy/ directory. 
+
+2. Ask ART for a scratch build. Make sure you can install  a cluster with that build.
+
+```
+oc adm release new \
```

Comment:
This is not applicable to OLM based operators - can we have a note or pointer here on testing the scratch build with OLM?

Reply:
Added some wording for this.